grading/multilang/grading/graders.py

```diff
@@ -12,217 +13,79 @@
-    """
-    Runs the script in file_name and returns a tuple (GraderResult, debug_info) depending on the
-    output for input_file_name. The second item of the tuple (debug_info) is a dictionary that
-    includes stdout, stderr, return_code and diff.
-
-    code: A string with the code to run.
-    runner: A CodeRunner instance.
-    input_file_name: The test case input file name.
-    expected_output_file_name: The test case output file name
-    options: An optional dictionary with grader settings. See grade_with_partial_scores() for
-        details.
-    """
-
-    if options is None:
-        options = {}
-
-    compute_diff = options.get("compute_diff", True)
-    diff_max_lines = options.get("diff_max_lines", 100)
-    diff_context_lines = options.get("diff_context_lines", 3)
-    check_output = options.get("check_output", _check_output)
-    treat_non_zero_as_runtime_error = options.get("treat_non_zero_as_runtime_error", True)
-
-    return_code, stdout, stderr = None, None, None
-    with open(input_file_name, 'r') as input_file:
-        return_code, stdout, stderr = project.run(input_file)
-
-    expected_output = None
-
-    with open(expected_output_file_name) as expected_output_file:
-        expected_output = expected_output_file.read()
-
-    result = None
-
-    if return_code == 0 or (not treat_non_zero_as_runtime_error and parse_non_zero_return_code(return_code) == GraderResult.RUNTIME_ERROR):
-        output_matches = check_output(stdout, expected_output)
-        result = GraderResult.ACCEPTED if output_matches else GraderResult.WRONG_ANSWER
-    else:
-        result = parse_non_zero_return_code(return_code)
-
-    debug_info = {}
-
-    if result != GraderResult.ACCEPTED:
-        diff = None
-        if compute_diff:
-            diff = _compute_diff(stdout, expected_output, diff_context_lines, diff_max_lines)
-
-        debug_info.update({
-            "input_file": input_file_name,
-            "stdout": html.escape(stdout),
-            "stderr": html.escape(stderr),
-            "return_code": return_code,
-            "diff": None if diff is None else html.escape(diff),
-        })
-
-    return result, debug_info
-
-def _compute_feedback(project, test_cases, options):
+class UserRequest:
     """
-    Computes the grader feedback for the given project against each of the provided test cases.
-
-    project: The project to run the test cases against.
-    test_cases: A list of tuples (input_file_name, output_file_name) describing the cases the code
-        will be tested against.
-    options: A dictionary with the grader options. It will be forwarded to _compute_single_feedback().
+    A "parameter object" that owns all the parts from
+    a request.
     """
-    grader_results = []
-    debug_info = {}
-
-    try:
-        project.build()
+    def __init__(self, problem_id, language_name=None):
+        """
+        Identifies the parts from an request
```

Comment:
What do you mean with parts from a request?
```suggestion
        Identifies or parses the request's data, that data is like the action type (i.e. custom_input) and so on.
```